src/vs/editor/contrib/referenceSearch/common/referenceSearch.ts

```diff
@@ -39,4 +39,30 @@ export function provideReferences(model: IReadOnlyModel, position: Position, pro
 	});
 }
 
+export function provideExternalReferences(model: IReadOnlyModel, query: ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: IReferenceInformation[]) => void): TPromise<IReferenceInformation[]> {
+
+	// collect references from all providers
+	const promises = ExternalReferenceProviderRegistry.ordered(model).map(provider => {
+		return asWinJsPromise((token) => {
+			return provider.provideExternalReferences(query, hints, token, progress);
+		}).then(result => {
+			if (Array.isArray(result)) {
+				return <IReferenceInformation[]>result;
+			}
+		}, err => {
+			onUnexpectedExternalError(err);
+		});
+	});
+
+	return TPromise.join(promises).then(references => {
+		let result: IReferenceInformation[] = [];
+		for (let ref of references) {
+			if (ref) {
+				result.push(...ref);
```

Comment:
65k would be a ridiculous number of references to send back, but I think it is a good idea to be defensive.